src/basic.h

```diff
@@ -61,8 +61,10 @@ class Symbol;
 */
 
 enum TypeID {
-    SYMBOL, MUL, ADD, POW, LOG,
-    NUMBER, INTEGER, RATIONAL, COMPLEX, CONSTANT,
+    INTEGER, RATIONAL, COMPLEX, REAL_DOUBLE, NUMBER,
+    // 'NUMBER' returns the number of subclasses of Number.
+    // All subclasses of Number must be added before it.
```

Comment:
Is this only used in `is_a_Number`? If so, perhaps we can find a different solution.